### DISCUSSION TITLE
New Attribute Similar to CallerMemberName but for Extension Method Source

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/csharplang#4206] [Language Ideas] [answered]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/566788/new-attribute-similar-to-callermembername-but-for.html)._

---
<p>I would love an attribute I could place on a string argument to have the compiler pass the name of the variable an extension method is being run on (i.e the item marked as 'this').</p> <p>Pseudo code for one example of of usage;</p> <pre>public static void GuardNull(this object argument, [ExtensionMethodCallerName] argumentName = &quot;&quot;)
{
	if (argument == null) throw new ArgumentNull(argumentName);
}

// When called like this...
myArgument.GuardNull();

// the arugmentName value passed to the extension method should contain &quot;myArgument&quot;

</pre>

---
#### Jane Wu [MSFT] on 5/14/2019, 01:51 AM: 

<p>Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further.&nbsp;We'll provide an update once the issue has been triaged by the product team.</p>

**Comment by HaloFour** [comment, marked as answer]
Should this be covered by https://github.com/dotnet/csharplang/issues/287 ?

**Comment by jerviscui** [comment]
Need it. 
Greate!

**Reply by 333fred** [reply]
Well, C# 10 will bring CallerArgumentExpression. So you'll have it.

**Reply by jerviscui** [reply]
> Well, C# 10 will bring CallerArgumentExpression. So you'll have it.

Thank you. 
I'm looking forward to it.
